src/fswAlgorithms/effectorInterfaces/stepperMotor/_UnitTest/test_stepperMotor.py

```diff
@@ -0,0 +1,180 @@
+    motorStepCountLog = StepperMotor.motorStepCountOutMsg.recorder(testProcessRate)
+    unitTestSim.AddModelToTask(unitTaskName, motorStepCountLog)
+
+    # Initialize the simulation
+    unitTestSim.InitializeSimulation()
+
+    # Calculate required number of steps for validation
+    if (initialMotorAngle > 0):
+        trueNumSteps = (desiredMotorAngle - (np.ceil(initialMotorAngle/stepAngle)*stepAngle)) / stepAngle
+    else:
+        trueNumSteps = (desiredMotorAngle - (np.floor(initialMotorAngle/stepAngle)*stepAngle)) / stepAngle
+
+    # If the desired motor angle is not a multiple of the step angle, the number of steps calculated is not an integer
+    # and it must be rounded to the nearest whole step
+    lowerStepFraction = trueNumSteps - np.floor(trueNumSteps)
+    upperStepFraction = np.ceil(trueNumSteps) - trueNumSteps
+    if (upperStepFraction > lowerStepFraction):
+        trueNumSteps = np.floor(trueNumSteps)
+    else:
+        trueNumSteps = np.ceil(trueNumSteps)
+
+    # If the desired motor angle is not a multiple of the step angle, a new desired angle is calculated
+    newMotorDesiredAngle = initialMotorAngle + (trueNumSteps * stepAngle)
+
+    # Set the simulation time
+    actuateTime = stepTime * np.abs(trueNumSteps)  # [sec] Time for the motor to actuate to the desired angle
+    holdTime = 5  # [sec] Time the simulation will continue while holding the final angle
+    unitTestSim.ConfigureStopTime(macros.sec2nano(actuateTime + holdTime))
+
+    # Begin the simulation
+    unitTestSim.ExecuteSimulation()
+
+    # Pull the logged motor step data
+    stepsCommanded = motorStepCountLog.stepsCommanded
+
+    # Check that the correct number of steps was calculated
+    if (stepsCommanded[0] != trueNumSteps):
+        testFailCount += 1
+        testMessages.append("\nFAILED: " + StepperMotorWrap.ModelTag + " Number of required motor steps do not match")
+
+    # Check to make sure the message was written correctly
+    if (trueNumSteps != 0):
+        if (np.count_nonzero(stepsCommanded) != 1):
+            testFailCount += 1
+            testMessages.append("\nFAILED: " + StepperMotorWrap.ModelTag + " MotorStepCountMsg was incorrectly written \nNum nonzero: " + str(np.count_nonzero(stepsCommanded)))
+    else:
+        if (np.count_nonzero(stepsCommanded) != 0):
+            testFailCount += 1
+            testMessages.append("\nFAILED: " + StepperMotorWrap.ModelTag + " MotorStepCountMsg was incorrectly written")
+
+    return [testFailCount, ''.join(testMessages)]
+
+
+# This statement below ensures that the unitTestScript can be run as a
+# stand-along python script
+if __name__ == "__main__":
+    stepperMotorTestFunction(
+                 False,
+                 1 * (np.pi / 180),     # stepAngle
+                 1,                     # stepTime
+                 0,                     # initialAngle
+                 10 * (np.pi / 180),    # desiredAngle
+               )
+    
```

Comment:
File is looking for a new line at the end.

Reply:
I think I fixed this..?

Reply:
yes